src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -2,18 +2,49 @@
 
 import core.basesyntax.Storage;
 
+@SuppressWarnings("unchecked")
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private Pair<K, V>[] pairs = new Pair[MAX_ITEMS_NUMBER];
+    private int size;
+
+    private class Pair<K,V> {
+        private final K key;
+        private V value;
+
+        public Pair(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
     @Override
     public void put(K key, V value) {
+        Pair pair = getPairByKey(key);
+        if (pair == null) {
+            pairs[size++] = new Pair<>(key, value);
+        } else {
+            pair.value = value;
+        }
+    }
+
+    private Pair getPairByKey(K key) {
```

Comment:
```suggestion
    private Pair<K, V> getPairByKey(K key) {
```
If you use generic types, don't use raw types